src/site/content/en/blog/granular-chunking-nextjs/index.md

```diff
@@ -0,0 +1,274 @@
+---
+title: Granular chunking in Next.js and Gatsby
+subhead: A newer webpack chunking strategy in Next.js and Gatsby minimizes duplicate code to improve page load performance 
+authors:
+  - houssein
+date: 2020-04-09
+hero: hero.png
+alt: Jigsaw pieces connecting 
+description: | 
+  Learn how both Next.js and Gatsby have improved their build output to minimize duplicate code and improve page load performance
+tags:
+  - post
+  - performance
+  - frameworks
+---
+
+Chrome is [collaborating](https://web.dev/advancing-framework-ecosystem-cds-2019/) with tooling and
+frameworks in the JavaScript open-source ecosystem. A number of newer optimizations were recently
+added to improve the loading performance of [Next.js](https://nextjs.org/) and
+[Gatsby](https://www.gatsbyjs.org/), and this article covers an improved granular chunking strategy
+that is now shipped by default in both frameworks.
+
+## Introduction
+
+Like many web frameworks, Next.js and Gatsby use [webpack](https://webpack.js.org/) as their core
+bundler. webpack v3 introduced
+[`CommonsChunkPlugin`](https://webpack.js.org/plugins/commons-chunk-plugin/) to make it possible to
+output modules shared between different entry points in a single (or few) "commons" chunk. Shared
+code can be downloaded separately and stored in the browser cache early on which can result in a
+better loading performance.
+
+This pattern became popular with many single-page application frameworks adopting an entrypoint and
+bundle configuration that looked like this:
+
+<img class="w-screenshot w-screenshot--filled" src="./commons-pattern-diagram.png" alt="Common entrypoint and bundle configuration">
+
+Although practical, the concept of bundling all shared module code into a single chunk has its
+limitations. Modules not shared in every entry point can be downloaded for routes that do not use it
+resulting in more code being downloaded than necessary (`moduleC` and `page1` in the diagram). For
+this reason, along with a few others, webpack v4 removed the plugin in favor of a new
+one: [`SplitChunksPlugin`](https://webpack.js.org/plugins/split-chunks-plugin/).
+
+## Improved Chunking
+
+The default settings for `SplitChunksPlugin`  works well for most users. Multiple split chunks are
+created depending on a number of [conditions](https://webpack.js.org/plugins/split-chunks-plugin/#defaults)
+to prevent fetching duplicated code across multiple routes. 
+
+However, many web frameworks that use this plugin still follow a "single-commons" approach to chunk
+splitting. Next.js, for example, would generate a `commons` bundle that contained any module that is
+used in more than 50% of pages and all framework dependencies (`react`, `react-dom`, etc…).
+
+```javascript/11,16
+const splitChunksConfigs: {
+  [propName: string]: webpack.Options.SplitChunksOptions
```

Comment:
It would be best to remove the TS typings from this, also cause there's no syntax highlighter for it.

Reply:
Removed 👍